src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { getPossiblePagesPagination }
+  from '../../functions/getPossiblePagesPagination';
+
+import './Pagination.scss';
+
+type Props = {
+  total: number;
+  perPage: number;
+  page: number;
+  onPageChange: (index: number) => void;
+  previousPage: () => void;
+  nextPage: () => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage = 5,
+  page = 1,
+  onPageChange,
+  previousPage,
+  nextPage,
+}) => {
+  const [isNextDisabled, setIsNextDisabled] = useState(false);
+  const [isPrevDisabled, setisPrevDisabled] = useState(false);
+
+  const currentPage = page;
+
+  const lastPage = Math.ceil(total / perPage);
+
+  const firstShowedElement = perPage * (currentPage - 1) + 1;
+  const lastShowedElement = currentPage * perPage > total
+    ? total
+    : currentPage * perPage;
+
+  const displayedPages = useMemo(() => {
+    return getPossiblePagesPagination(currentPage, lastPage);
+  }, [total, perPage, page]);
+
+  useEffect(() => {
+    if (page > 1) {
+      setisPrevDisabled(false);
+    } else {
+      setisPrevDisabled(true);
+    }
```

Comment:
you can create const isPrevDisabled where with useMemo check page number

`
const isPrevDisabled = useMemo(() => page === 1, [page]);
`

Reply:
noted